### ISSUE TITLE
epam does not stay root when invoked by ejabberd

### ISSUE THREAD

**Issue by jwise** [jabber-at/ejabberd#15]
> What version of ejabberd are you using?

```
root@moroso:~# dpkg --status ejabberd
Package: ejabberd
[...]
Maintainer: Mathias Ertl <mati@jabber.at>
Architecture: amd64
Version: 1:16.09-1~afa80
```

> What operating system (version) are you using?

```
root@moroso:~# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 8.7 (jessie)
Release:        8.7
Codename:       jessie
```

> How did you install ejabberd (source, package, distribution)?

from apt; `deb https://apt.jabber.at jessie ejabberd`

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

I recently upgraded my machine (to 16.09; I'm not sure what the old version was), and now `epam` does not stay running as root, for some reason, and authentication fails when I try to log in to my Jabber server.  First off, let's look at the permissions:

```
root@moroso:~# ps aux | grep epam
ejabberd 10864  0.0  0.0  12804   924 ?        Ss   01:47   0:00 /usr/lib/erlang/p1_pam/bin/epam
root@moroso:~# ls -l /usr/lib/erlang/p1_pam/bin/epam
-rwsr-xr-- 1 root ejabberd 55176 Jan 16  2016 /usr/lib/erlang/p1_pam/bin/epam*
```

I'll kill epam, to show that it's not just an old epam running around from before the epam binary was suid root:

```
root@moroso:~# kill `pidof epam`
root@moroso:~# ps aux | grep epam
ejabberd 11149  0.0  0.0  12804   976 ?        Ss   01:57   0:00 /usr/lib/erlang/p1_pam/bin/epam
```

(note PID change)

But, even more strangely, if I invoke epam by hand as the ejabberd user, it *does* stay running as root!

```
root@moroso:~# su ejabberd
ejabberd@moroso:/root$ /usr/lib/erlang/p1_pam/bin/epam
```

(in another terminal)

```
root@moroso:~# ps aux | grep epam
ejabberd 11149  0.0  0.0  12804   976 ?        Ss   01:57   0:00 /usr/lib/erlang/p1_pam/bin/epam
root     11203  0.0  0.0  12804  1008 pts/15   S+   01:57   0:00 /usr/lib/erlang/p1_pam/bin/epam
```

Note that this time it came up as root!

Any idea what's going on here?  I am feeling pretty stuck here.

Thanks,
joshua

**Comment by jwise** [comment]
Ok, I had a hunch this was capability-related or systemd-related, and indeed, it is.  When I `systemctl stop ejabberd`, and `ejabberdctl start`, then epam happily comes up as root.  But if I use ejabberd through systemd, then epam is stuck as ejabberd.

The systemd `ejabberd.service` has `NoNewPrivileges=true` in it, which is what's preventing epam from escalating.  When I comment that out, epam starts perfectly happily.

**Comment by mathiasertl** [comment]
I confirm this. The line comes from upstream. It was removed somewhere between 16.09 and 17.01. The problem will be solved as soon as I manage to build 17.01 packages!
